### ISSUE TITLE
shadowsocks{,-libev} and haveged: The shadowsocks systemd service is likely to start before haveged

### ISSUE THREAD

**Issue by eatradish** [AOSC-Dev/aosc-os-abbs#1912]
Bug description
-----------------------

If the system is low on entropy, one would need to start `haveged.service` before `shadowsocks{,-libev} .service`. If both services are enabled, as both are `WantedBy=multi-user.target`, `shadowsocks{,-libev} .service` may start before `haveged.service`. In this case, `shadowsocks{,-libev} .service` may fail as `haveged.service` has not generated enough entropy.

Update channel(s)
--------------------------

All

**Comment by LionNatsu** [comment]
Could you change WantedBy= from multi-user.target to basic.target and try again?

**Comment by eatradish** [comment]
> Could you change WantedBy= from multi-user.target to basic.target and try again?

It seems that the problem has not been solved.

```
⋊> ~ sudo systemctl cat haveged                                                                                                                                   17:52:46
# /usr/lib/systemd/system/haveged.service
[Unit]
Description=Entropy Harvesting Daemon
Documentation=man:haveged(8)

[Service]
ExecStart=/usr/bin/haveged -F -w 1024 -v 63
SuccessExitStatus=143

[Install]
WantedBy=basic.target

⋊> ~ sudo systemctl status haveged                                                                                                                                17:53:22
● haveged.service - Entropy Harvesting Daemon
   Loaded: loaded (/usr/lib/systemd/system/haveged.service; enabled; vendor preset: disabled)
   Active: active (running) since Sat 2019-07-13 17:51:03 CST; 2min 33s ago
     Docs: man:haveged(8)
 Main PID: 1111 (haveged)
    Tasks: 1 (limit: 4915)
   Memory: 3.8M
   CGroup: /system.slice/haveged.service
           └─1111 /usr/bin/haveged -F -w 1024 -v 63

Jul 13 17:51:03 eat-x220i haveged[1111]: haveged: Loop 40: offset=0, delta=1152
Jul 13 17:51:04 eat-x220i haveged[1111]: haveged: AIS-31 tot procedure B: success 873346 bytes fill 0
Jul 13 17:51:04 eat-x220i haveged[1111]: haveged: procedure B: test6a:1/1, test6b:1/1, test7a:2/2, test7b:4/4, test8:1/1
Jul 13 17:51:04 eat-x220i haveged[1111]: haveged: AIS-31 tot procedure A: success 1035716 bytes fill 0
Jul 13 17:51:04 eat-x220i haveged[1111]: haveged: procedure A: test0:1/1, test1:257/257, test2:257/257, test3:257/257, test4:257/257, test5:2/2
Jul 13 17:51:04 eat-x220i haveged[1111]: haveged: ver: 1.9.2; arch: x86; vend: GenuineIntel; build: (gcc 7.3.1 ITV); collect: 128K
Jul 13 17:51:04 eat-x220i haveged[1111]: haveged: cpu: (L4 VC); data: 32K (L4 V); inst: 32K (L4 V); idx: 22/40; sz: 32070/59219
Jul 13 17:51:04 eat-x220i haveged[1111]: haveged: tot tests(BA8): A:1/1 B:1/1 continuous tests(B):  last entropy estimate 8.00001
Jul 13 17:51:04 eat-x220i haveged[1111]: haveged: fills: 0, generated: 0
Jul 13 17:51:04 eat-x220i haveged[1111]: haveged: 0 fill 20.0271 ms

⋊> ~ systemctl status shadowsocks-libev@config                                                                                                                    17:52:00
● shadowsocks-libev@config.service - Shadowsocks-Libev Client Service
   Loaded: loaded (/usr/lib/systemd/system/shadowsocks-libev@.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Sat 2019-07-13 17:51:04 CST; 1min 4s ago
  Process: 1247 ExecStart=/usr/bin/ss-local -c /etc/shadowsocks/config.json (code=exited, status=255/EXCEPTION)
 Main PID: 1247 (code=exited, status=255/EXCEPTION)

Jul 13 17:51:04 eat-x220i systemd[1]: Started Shadowsocks-Libev Client Service.
Jul 13 17:51:04 eat-x220i ss-local[1247]:  2019-07-13 17:51:04 INFO: initializing ciphers... chacha20-ietf-poly1305
Jul 13 17:51:04 eat-x220i ss-local[1247]:  2019-07-13 17:51:04 INFO: This system doesn't provide enough entropy to quickly generate high-quality random numbers.
Jul 13 17:51:04 eat-x220i ss-local[1247]: Installing the rng-utils/rng-tools, jitterentropy or haveged packages may help.
Jul 13 17:51:04 eat-x220i ss-local[1247]: On virtualized Linux environments, also consider using virtio-rng.
Jul 13 17:51:04 eat-x220i ss-local[1247]: The service will not start until enough entropy has been collected.
Jul 13 17:51:04 eat-x220i ss-local[1247]:  2019-07-13 17:51:04 ERROR: getaddrinfo: Temporary failure in name resolution
Jul 13 17:51:04 eat-x220i ss-local[1247]:  2019-07-13 17:51:04 ERROR: failed to resolve the provided hostname
Jul 13 17:51:04 eat-x220i systemd[1]: shadowsocks-libev@config.service: Main process exited, code=exited, status=255/EXCEPTION
Jul 13 17:51:04 eat-x220i systemd[1]: shadowsocks-libev@config.service: Failed with result 'exit-code'.
```

**Comment by eatradish** [comment]
Fix with #1912, Closing.
